### PR TITLE
Add download description for Windows nightly build

### PR DESCRIPTION
**Pull request by terrychan999**
This is a preliminary PR for https://github.com/solvcon/modmesh/issues/371
The url query string is helpful for filtering the successful nightly build on master branch:
```
?query=event%3Aschedule+is%3Asuccess+branch%3Amaster
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,4 +3,9 @@
 Check the Github Actions workflow for how to build the code.
 Check https://github.com/solvcon/mmnote for the theoretical and numerical notes.
 
+## Binaries
```
Comment (yungyuc) [inline README.md]: Please be specific that this is for Windows binary.  "Windows portable binaries" reads clearer.

File: README.md [hunk 2]
```diff
@@ -3,4 +3,9 @@
...
+A portable version of the modmesh viewer for Windows is now available for download from the Actions: [**devbuild**](https://github.com/solvcon/modmesh/actions/workflows/devbuild.yml?query=event%3Aschedule+is%3Asuccess+branch%3Amaster) (login required).
```
Comment (yungyuc) [inline README.md]: Please add a blank line between the section and the paragraph.

Please follow the plain text formatting suggested by the vim modeline at the bottom of the file.  Try to make the line width to be smaller than 80 characters as much as possible.

**Review by yungyuc — CHANGES_REQUESTED** [review]
Thanks, @terrychan999 .  This looks good.  We should use a little bit of modifications.

- [ ] More specific section title.
- [ ] Formatting: blank lines and line width.

File: README.md (hunk 1 above)
Comment (tigercosmos) [reply README.md]: personally, I prefer "Download", which is more common.

Comment (yungyuc) [reply README.md]: Download sounds good

**Comment by yungyuc** [comment]
@j8xixo12 and @YenPeiChen07 could you please help take a look too?

File: README.md [hunk 3]
```diff
@@ -3,4 +3,9 @@
...
+
+> Click on a run result and scroll down to the Artifacts section to download the ZIP file.
```
Comment (j8xixo12) [inline README.md]: If add a clearly indicating to click `build_windows-2022_Release` and `archive_portable_artifacts` should help users find the download url more quickly.

Comment (terrychan999) [reply README.md]: I'm not sure if a two-step click from the job `build_windows-2022_Release` to the step `archive_portable_artifacts` is more intuitive than scrolling down to the Artifacts section with just one click in the workflow.

**Comment by terrychan999** [comment]
What about generating a single click download link:

I have written a cool script that helps us redirect to the latest link of the artifacts.
However, it requires enabling GitHub Pages because we cannot embed JavaScript directly in `README.md`.

Here is an example download link:
https://terrychan999.github.io/modmesh/download

Here is the source code:
https://github.com/terrychan999/modmesh/blob/4a3840815e985e592b261c7f8c10f95b24cf2e9d/docs/download.html

This does not require API tokens, and all the behavior is conducted on the client side, so we don't have security and privacy risks.

**Comment by yungyuc** [comment]
> I have written a cool script that helps us redirect to the latest link of the artifacts. However, it requires enabling GitHub Pages because we cannot embed JavaScript directly in `README.md`.
> 
> Here is an example download link: https://terrychan999.github.io/modmesh/download

It is cool!  I'd like to keep the modmesh repository to be code-major.  We can use a separate repository for the documents and download links: https://github.com/solvcon/solvcon-doc .  Could you please create a PR at the [solvcon-doc](https://github.com/solvcon/solvcon-doc) repository?

**Comment by yungyuc** [comment]
@terrychan999 Any updates?

**Comment by terrychan999** [comment]
> @terrychan999 Any updates?

Hi, I will file a PR about download links to [solvcon-doc](https://github.com/solvcon/solvcon-doc) in this Wednesday(7/10), probably during sciwork meetup.

**Comment by yungyuc** [comment]
> > @terrychan999 Any updates?
> 
> Hi, I will file a PR about download links to [solvcon-doc](https://github.com/solvcon/solvcon-doc) in this Wednesday(7/10), probably during sciwork meetup.

Thanks for updating with a plan.

**Comment by yungyuc** [comment]
@terrychan999 is there progress on this?

**Comment by terrychan999** [comment]
> @terrychan999, is there any progress on this?

I've created a pull request at https://github.com/solvcon/solvcon-doc/pull/1. I think it can be reviewed first. Thanks!

**Review by yungyuc — APPROVED** [review]
LGTM.  Thanks, @terrychan999 .

**Comment by yungyuc** [comment]
Oops.  It seems that I accidentally merged this PR.  I have to swallow the mistake.  @terrychan999 could you please create a new PR for updating `README.md` with the new download link?

**Comment by yungyuc** [comment]
> Oops. It seems that I accidentally merged this PR. I have to swallow the mistake. @terrychan999 could you please create a new PR for updating `README.md` with the new download link?

I've done the fix.  See https://github.com/solvcon/modmesh/pull/393

**Pull request merged**